src/commands/components/index.ts

```diff
@@ -0,0 +1,64 @@
+import Command from '../../base-command';
+import Table from '../../base-table';
+import { Account, AccountUtils } from '../../common/utils/account';
+import localizedTimestamp from '../../common/utils/localized-timestamp';
+
+interface Component {
+  created_at: string;
+  updated_at: string;
+  name: string;
+  metadata: {
+    tag_count: number;
+  };
+  account: Account;
+}
+
+export default class Components extends Command {
+  static aliases = ['components', 'components:search', 'component:search', 'component:search'];
+  static description = 'Search components you have access to';
+
+  static flags = {
+    ...Command.flags,
+    ...AccountUtils.flags,
+  };
+
+  static args = [{
+    name: 'query',
+    description: 'Search term used to filter the results',
+  }];
+
+  async run(): Promise<void> {
+    const { args, flags } = this.parse(Components);
+
+    let account: Account | undefined = undefined;
+    if (flags.account) {
+      account = await AccountUtils.getAccount(this.app.api, flags.account);
+    }
+
+    const params = {
+      q: args.query || '',
+      account_id: account?.id,
+    };
+
+    let { data: { rows: components } } = await this.app.api.get(`/components`, { params });
+    components = components.filter((c: Component) => c.account);
+
+    if (!components.length) {
+      this.log('You have not registered any components yet.');
+      return;
+    }
+
+    const table = new Table({ head: ['Name', 'Account', 'Versions', 'Created', 'Updated'] });
+    for (const component of components.sort((c1: Component, c2: Component) => c1.account.name.localeCompare(c2.account.name)).sort((c1: Component, c2: Component) => c1.name.localeCompare(c2.name))) {
```

Comment:
Is the intent here to sort the components by name, but grouped by account? If so, I think you need to reverse the order these sorts are applied, since this is splitting up the account groups:
```
% architect-local components
┌─────────────────────────────┬───────────────┬──────────┬───────────────────────────┬──────────────────────────┐
│ Name                        │ Account       │ Versions │ Created                   │ Updated                  │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ cloud                       │ architect     │ 1        │ 9/13/21, 3:39:07 PM PDT   │ 9/13/21, 3:39:07 PM PDT  │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ gcp-pubsub                  │ architect     │ 1        │ 10/13/21, 10:18:31 AM PDT │ 10/13/21, 3:47:07 PM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ ideaboards                  │ brahm-testing │ 1        │ 8/20/21, 10:33:41 AM PDT  │ 9/22/21, 10:37:13 AM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ ideaboards-backend-notifier │ brahm-testing │ 1        │ 8/20/21, 10:32:34 AM PDT  │ 8/20/21, 10:32:34 AM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ ideaboards-backend-rest     │ brahm-testing │ 1        │ 8/20/21, 10:32:51 AM PDT  │ 8/20/21, 10:32:51 AM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ ideaboards-frontend         │ brahm-testing │ 1        │ 8/20/21, 10:03:06 AM PDT  │ 8/20/21, 10:03:06 AM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ my-task                     │ brahm-testing │ 5        │ 9/22/21, 9:39:32 AM PDT   │ 10/12/21, 4:12:47 PM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ postgres                    │ architect     │ 2        │ 8/20/21, 10:28:54 AM PDT  │ 8/23/21, 12:00:14 PM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ publisher                   │ examples      │ 1        │ 10/13/21, 10:18:42 AM PDT │ 10/14/21, 8:47:04 PM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ react-app                   │ examples      │ 1        │ 8/20/21, 9:58:11 AM PDT   │ 8/20/21, 9:58:11 AM PDT  │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ redis                       │ brahm-testing │ 1        │ 8/20/21, 9:51:53 AM PDT   │ 8/20/21, 9:51:53 AM PDT  │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ repro-interpolation-bug     │ brahm-testing │ 1        │ 9/13/21, 9:07:38 AM PDT   │ 9/13/21, 9:07:38 AM PDT  │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ subscriber                  │ examples      │ 1        │ 10/13/21, 10:18:56 AM PDT │ 10/14/21, 8:47:20 PM PDT │
└─────────────────────────────┴───────────────┴──────────┴───────────────────────────┴──────────────────────────┘
```

Reply:
```suggestion
    const sorted_components = components.sort((c1: Component, c2: Component) => c1.name.localeCompare(c2.name)).sort((c1: Component, c2: Component) => c1.account.name.localeCompare(c2.account.name));
    for (const component of sorted_components) {
```

This seems to work:
```
% architect-local components       
┌─────────────────────────────┬───────────────┬──────────┬───────────────────────────┬──────────────────────────┐
│ Name                        │ Account       │ Versions │ Created                   │ Updated                  │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ cloud                       │ architect     │ 1        │ 9/13/21, 3:39:07 PM PDT   │ 9/13/21, 3:39:07 PM PDT  │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ gcp-pubsub                  │ architect     │ 1        │ 10/13/21, 10:18:31 AM PDT │ 10/13/21, 3:47:07 PM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ postgres                    │ architect     │ 2        │ 8/20/21, 10:28:54 AM PDT  │ 8/23/21, 12:00:14 PM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ ideaboards                  │ brahm-testing │ 1        │ 8/20/21, 10:33:41 AM PDT  │ 9/22/21, 10:37:13 AM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ ideaboards-backend-notifier │ brahm-testing │ 1        │ 8/20/21, 10:32:34 AM PDT  │ 8/20/21, 10:32:34 AM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ ideaboards-backend-rest     │ brahm-testing │ 1        │ 8/20/21, 10:32:51 AM PDT  │ 8/20/21, 10:32:51 AM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ ideaboards-frontend         │ brahm-testing │ 1        │ 8/20/21, 10:03:06 AM PDT  │ 8/20/21, 10:03:06 AM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ my-task                     │ brahm-testing │ 5        │ 9/22/21, 9:39:32 AM PDT   │ 10/12/21, 4:12:47 PM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ redis                       │ brahm-testing │ 1        │ 8/20/21, 9:51:53 AM PDT   │ 8/20/21, 9:51:53 AM PDT  │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ repro-interpolation-bug     │ brahm-testing │ 1        │ 9/13/21, 9:07:38 AM PDT   │ 9/13/21, 9:07:38 AM PDT  │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ publisher                   │ examples      │ 1        │ 10/13/21, 10:18:42 AM PDT │ 10/14/21, 8:47:04 PM PDT │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ react-app                   │ examples      │ 1        │ 8/20/21, 9:58:11 AM PDT   │ 8/20/21, 9:58:11 AM PDT  │
├─────────────────────────────┼───────────────┼──────────┼───────────────────────────┼──────────────────────────┤
│ subscriber                  │ examples      │ 1        │ 10/13/21, 10:18:56 AM PDT │ 10/14/21, 8:47:20 PM PDT │
└─────────────────────────────┴───────────────┴──────────┴───────────────────────────┴──────────────────────────┘
```

Reply:
Oh yeah I definitely agree with that- sorting by just the name is likely the expected behavior here 👍